workspace_tools/test_api.py

```diff
@@ -479,17 +518,17 @@ def execute_thread_slice(self, q, target, toolchains, clean, test_ids, build_rep
                 test_suite_properties['test.path.%s.%s.%s'% (target, toolchain, test_id)] = path
 
                 # read MUTs, test specification and perform tests
-                single_test_result, detailed_test_results = self.handle(test_spec, target, toolchain, test_loops=test_loops)
+                handle_result = self.handle(test_spec, target, toolchain, test_loops=test_loops)
```

Comment:
This will fix:
```
[DEBUG] Command: C:/mbed_tools/ARMCompiler_5.03_117_Windows\bin\fromelf --bin -o C:\Jenkins\jobs\mbed_2.0_build_release_test_publish\workspace\build\test\NRF51822\ARM\DTCT_1\detect.hex C:\Jenkins\jobs\mbed_2.0_build_release_test_publish\workspace\build\test\NRF51822\ARM\DTCT_1\detect.elf
[DEBUG] Return: 0
[DEBUG] Merge SoftDevice file s110_nrf51822_8.0.0_softdevice.hex
Traceback (most recent call last):
  File "./workspace_tools/singletest.py", line 237, in <module>
    singletest_in_cli_mode(single_test)
  File "C:\Jenkins\jobs\mbed_2.0_build_release_test_publish\workspace\workspace_tools\test_api.py", line 1419, in singletest_in_cli_mode
    test_summary, shuffle_seed, test_summary_ext, test_suite_properties_ext, build_report = single_test.execute()
  File "C:\Jenkins\jobs\mbed_2.0_build_release_test_publish\workspace\workspace_tools\test_api.py", line 543, in execute
    self.execute_thread_slice(q, target, toolchains, clean, test_ids, cur_build_report)
  File "C:\Jenkins\jobs\mbed_2.0_build_release_test_publish\workspace\workspace_tools\test_api.py", line 482, in execute_thread_slice
    single_test_result, detailed_test_results = self.handle(test_spec, target, toolchain, test_loops=test_loops)
TypeError: 'NoneType' object is not iterable
``` 
when None object is passed as port or mount point from e.g. mbed-ls tools